packages/database/src/modelClasses/Dashboard.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { DatabaseModel } from '../DatabaseModel';
+import { DatabaseType } from '../DatabaseType';
+import { TYPES } from '../types';
+
+class DashboardType extends DatabaseType {
+  static databaseType = TYPES.DASHBOARD;
+}
+
+export class DashboardModel extends DatabaseModel {
+  get DatabaseTypeClass() {
+    return DashboardType;
+  }
+
+  async getDashboards(entity, hierarchyId, projectCode, params) {
+    const ancestorCodes = await entity.getAncestorCodes(hierarchyId);
```

Comment:
This raises an interesting question about whether we want to continue allowing _any_ entity relationship access to exist outside of entity-server, or force this information to be fetched. It would probably fit better with our architecture to have the requester pass in the ancestors, rather than having the endpoint work that out. That does mean putting complexity in multiple clients rather than this one spot though.

This model is also structured really nicely to take care of accounting for the multiple semi-complicated filter rules in getting dashboards, which is exactly what a model should do.

Let's leave it how you've got it for now, but @rohan-bes it's a point for us to keep in mind about whether the entity-server should be the sole point of contact for entity relationship stuff. The benefit of that would be that we could pull it out into a proper micro-service that potentially even had its own database.

Reply:
Hmmm that's a good question @edmofro. So far, I've largely endeavoured to keep the 'entity relation' type logic that report-server deals with in the `EntityModel` (or `AncestorDescendantRelation` model) so that the entity-server itself isn't introducing new business rules about what relations exist between entities.

As such I think for now it's still fairly safe for code that has access to the `EntityModel` to query for relations and maintain consistent behaviour with the entity-server. This mightn't always be the case though...

The entity-server also handles permissions which the entity model does not... This may become even more apparent with our new 'Public data per application' system... I guess calls to `getAncestors()` should be safe as if you have a descendant already you should have access to all its ancestors, but the inverse is not true...